config/reader_test.go

```diff
@@ -394,3 +367,20 @@ func TestReadTLSFromCache(t *testing.T) {
 		test.That(t, err, test.ShouldBeNil)
 	})
 }
+
+func TestProcessConfigRegistersLogConfig(t *testing.T) {
```

Comment:
let's add a similar test that includes a service and a component config that overrides config in `LogConfig`